frontend/app/src/components/Sidebar/styled-components.ts

```diff
@@ -290,12 +333,16 @@ export const StyledCollapseSidebarButton =
   styled.div<StyledCollapseSidebarButtonProps>(
     ({ showSidebarCollapse, theme }) => {
       const isLightTheme = hasLightBackgroundColor(theme)
+      const isCustomTextColor = hasCustomTextColor(theme)
 
       return {
         display: showSidebarCollapse ? "inline" : "none",
         transition: "left 300ms",
         transitionDelay: "left 300ms",
-        color: isLightTheme ? theme.colors.gray70 : theme.colors.bodyText,
+        color:
+          isLightTheme && !isCustomTextColor
+            ? theme.colors.gray70
+            : theme.colors.bodyText,
```

Comment:
Seems like we should perhaps abstract this to a function as well? I can imagine losing track of this.